### DISCUSSION TITLE
Amazon Ads API rate limiter question

### DISCUSSION THREAD

**Discussion by JRseer** [amzn/ads-advanced-tools-docs#263] [Q&A]
the Create Report API documention: https://advertising.amazon.com/API/docs/en-us/offline-report-prod-3p#tag/Asynchronous-Reports/operation/createAsyncReport.

for error 429:  
1. what is the rate limits for a second?
2. any factors affacting the rate limits? report types、merchantId and ?
3. would there any difference between Amazon Ads API rate limits and SP-API's?

**Comment by xuqiang97** [comment]
we also need this help!

**Comment by elizafarley** [comment]
HI @JRseer and @xuqiang97, 
To answer questions 1 and 2 above, in general rate limits are dynamic based on traffic to the API. You can see more info here: https://advertising.amazon.com/API/docs/en-us/reference/concepts/rate-limiting#avoiding-rate-limits-for-reports.

For 3, the Ads API and SP-APIs are separate APIs with different throttling mechanisms. I don't work on the SP-API, but you might find these docs helpful: https://developer-docs.amazon.com/sp-api/docs/usage-plans-and-rate-limits-in-the-sp-api.

**Reply by JRseer** [reply]
hi elizafarley,
Thank you very much for your reply，it's because I couldn't find the desired information on [the rate limiting page](https://advertising.amazon.com/API/docs/en-us/reference/concepts/rate-limiting#avoiding-rate-limits-for-reports) that I raised the questions above.
Does the AD API rate limiting have a minimum allowed number of report creations? For example, when creating a Purchased Product report, during idle server times, it can process up to 5 report creations per second, while during busy server times, it can handle 2 report creations per second, with a minimum limit of processing 2 report creations per second. I need the actual parameters to control and reduce the occurrences of exceeding rate limits when using the Amazon Ads API.
Looking forward to your response, thanks

**Reply by siddharthdedhiaheadln** [reply]
Hey any updates on this?
